### PR TITLE
implement all functionality

### PR DESCRIPTION
**Pull request by CarolussRex**
@Nivolves

### CODE REVIEW

**Comment by CarolussRex** [comment]
@Nivolves

File: package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "@testing-library/jest-dom": "^5.14.1",
+    "@testing-library/react": "^11.2.7",
+    "@testing-library/user-event": "^12.8.3",
+    "node-sass": "^6.0.1",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-scripts": "4.0.3",
+    "sass": "^1.43.4",
+    "web-vitals": "^1.1.2",
+    "wowjs": "^1.1.3"
```
Comment (Nivolves) [inline package.json:15]: why do you need it?

File: src/App/App.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+ 
+  const deletePost = (post) => {
+  setList(list.filter(p => p.id !== post.id))
+  }
+
+  const changePost = (post, e) =>{
+      const filter = list.filter(p => p.id !== post.id) 
+      const find = list.find(p => p.id === post.id)
+      find.title = e.target.value
```
Comment (Nivolves) [inline src/App/App.js:31]: its a bad practice to change something in object 
const index = list.findIndex(({id}) => id === post.id);
setList(Object.assign([...list], {
 [index]: {
   ...list[index],
  title: e.target.value
 }
})

File: src/App/App.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  })
+
+  return(
+  <div className='main'>
+    <form>
+      <input type='text' placeholder='Type here' onChange={createPost} id="add-input" value={title}></input>
+      <button onClick={AddPost} id = "button-add">Add your Plan</button>
+    </form>
+      {list.map((item, k) =>
+            <ItemOfTheList key = {k}
```
Comment (Nivolves) [inline src/App/App.js:48]: u could use item.id like key

File: src/ItemOfTheList/ItemOfTheList.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React, { useState,} from 'react';
+
+export function ItemOfTheList(props) {
```
Comment (Nivolves) [inline src/ItemOfTheList/ItemOfTheList.js:3]:
```suggestion
export function ItemOfTheList({ item, change, deletting }) {
```

**Review by Nivolves — CHANGES_REQUESTED** [review]
Project should be on typescript. Use arrow functions as react components

**Pull request closed without merging**